ui/packages/atlasmap/src/UI/dialogs/ParametersDialog.tsx

```diff
@@ -0,0 +1,178 @@
+import React, {
+  FunctionComponent,
+  useState,
+  useCallback, useEffect,
+} from "react";
+
+import {
+  Button,
+  Form,
+  FormGroup, FormSelect, FormSelectOption, Split, SplitItem,
+  TextInput,
+} from "@patternfly/react-core";
+
+import {
+  ConfirmationDialog,
+  IConfirmationDialogProps,
+} from "./ConfirmationDialog";
+import {PlusIcon, TimesIcon} from "@patternfly/react-icons";
+
+export interface IParameter {
+  name: string;
+  value: string;
+  options?: IParameterOption[];
+  hidden?: boolean;
+  hideable?: boolean;
+}
+
+export interface IParameterOption {
+  label: string;
+  value: string;
+}
+
+export interface IParametersDialogProps {
+  title: string;
+  parameters?: IParameter[];
+  isOpen: IConfirmationDialogProps["isOpen"];
+  onCancel: IConfirmationDialogProps["onCancel"];
+  onConfirm: (parameters: IParameter[]) => void;
+}
+export const ParametersDialog: FunctionComponent<IParametersDialogProps> = ({
+  title,
+  parameters: initialParameters = [],
+  isOpen,
+  onCancel,
+  onConfirm,
+}) => {
+  const initialAddParameter = 'Select...';
+
+  const [parameters, setParameters] = useState(initialParameters);
+
+  const [addParameter, setAddParameter] = useState(initialAddParameter);
+
+  const reset = useCallback(() => {
+    setParameters(initialParameters);
+  }, [initialParameters]);
+
+  const handleOnConfirm = useCallback(() => {
+    onConfirm(parameters);
+  }, [parameters, onConfirm]);
+
+  const handleOnCancel = useCallback(() => {
+    onCancel();
+    reset();
+  }, [onCancel, reset]);
+
+  const handleAddParameter = useCallback(() => {
```

Comment:
Following the previous comment, this would be cleared if the selected parameter is an argument of the function, since all you'd have to do is to set the new array in the state.

Reply:
I'm sorry for my react ignorance. I thought I needed that to access value from the select field.

Reply:
I couldn't figure out how to pass it as an argument or I'm not following what you mean...

Reply:
Maybe I wasn't clear, but you did what I meant :D https://github.com/atlasmap/atlasmap/pull/2212/files#diff-a22ea50982c257cee5dbd972c2a46731R134-R139

Reply:
I thought you were asking to pass the value as argument to https://github.com/atlasmap/atlasmap/pull/2212/files#diff-a22ea50982c257cee5dbd972c2a46731R49 Anyway, glad to hear it's ok now.